src/main/java/com/faforever/api/user/UserController.java

```diff
@@ -26,8 +28,34 @@ public void register(@RequestParam("username") String username,
   }
 
   @ApiOperation("Activates a previously registered account.")
-  @RequestMapping(path = "/activate", method = RequestMethod.POST, produces = MediaType.APPLICATION_JSON_VALUE)
+  @RequestMapping(path = "/activate", method = RequestMethod.GET, produces = MediaType.APPLICATION_JSON_VALUE)
   public void activate(@RequestParam("token") String token) {
     userService.activate(token);
   }
+
+  @PreAuthorize("#oauth2.hasScope('change_password') and hasRole('ROLE_USER')")
+  @ApiOperation("Changes the password of a previously registered account.")
+  @RequestMapping(path = "/changePassword", method = RequestMethod.POST, produces = MediaType.APPLICATION_JSON_VALUE)
+  public void changePassword(@RequestParam("newPassword") String newPassword, Authentication authentication) {
+    userService.changePassword(newPassword, userService.getUser(authentication));
```

Comment:
Yes, but it should require the original password as a parameter. Oauth token alone is not safe .